src/database/schema/ChatSchema.ts

```diff
@@ -284,6 +297,19 @@ class ChatSchema implements BaseDbSchema {
     let rawJson: string | null = null;
     let attachmentJson: string | null = null;
 
+    let type = json?.type;
+    if (
+      json?.message_type === MESSAGE_TYPE_DELETED ||
+      json?.message_type === 'notification-deleted'
+    ) {
+      type = MESSAGE_TYPE_DELETED;
+    }
+
+    let message = json?.text || json?.message || '';
+    if (json?.message_type === MESSAGE_TYPE_DELETED) {
+      message = DELETED_MESSAGE_TEXT;
+    }
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

As mentioned above, the logic for checking if a message is deleted and setting the `type` and `message` variables is duplicated. Extracting this logic into a separate function would improve maintainability and readability of your code.



<!-- This is an auto-generated comment by OSS CodeRabbit -->